lib/webmock/util/hash_counter.rb

```diff
@@ -5,22 +5,36 @@
 module WebMock
   module Util
     class HashCounter
-      attr_accessor :hash
+      attr_accessor :hash, :array
 
       def initialize
         self.hash = Hash.new(0)
         @order = {}
         @max = 0
         @lock = ::Mutex.new
+        self.array = []
+        @request_object_ids = {}
       end
 
       def put(key, num=1)
         @lock.synchronize do
+          store_to_array(key, num)
           hash[key] += num
           @order[key] = @max += 1
         end
       end
 
+      def store_to_array(key, num)
+        request_object_id = @request_object_ids[key]
+        request_object_id = key.object_id if request_object_id.nil?
+        num.times do
+          array << ObjectSpace._id2ref(request_object_id)
+        rescue RangeError
```

Comment:
@johngallagher have you experienced a scenario where that `rescue` is needed? when can that happen?

Reply:
Let's take your advice and simplify this code! It can happen when an old reference is kept but it's out of bounds. This happened once or twice when running the test suite - I assumed it's because the references hadn't been cleaned up properly but never dug into it.

I didn't really like doing this, so anything that's simpler and doesn't use this weird `#_id2ref` method has my vote!

Reply:
OK, simplified as you suggested and it passes all the tests and seems to work! Nice.